presto-main/src/main/java/com/facebook/presto/operator/DynamicFilterClient.java

```diff
@@ -0,0 +1,99 @@
+import static io.airlift.http.client.StatusResponseHandler.createStatusResponseHandler;
+import static io.airlift.json.JsonCodec.jsonCodec;
+import static java.util.Objects.requireNonNull;
+
+public class DynamicFilterClient
+        implements Closeable
+{
+    private final JsonCodec<DynamicFilterSummary> summaryJsonCodec;
+    private final URI coordinatorURI;
+    private final HttpClient httpClient;
+    private final TaskId taskId;
+    private final String source;
+    private final int driverId;
+    private final int expectedDriversCount;
+
+    public DynamicFilterClient(JsonCodec<DynamicFilterSummary> summaryJsonCodec, URI coordinatorURI, HttpClient httpClient, TaskId taskId, String source, int driverId, int expectedDriversCount)
+    {
+        this.summaryJsonCodec = requireNonNull(summaryJsonCodec, "summaryJsonCodec is null");
+        this.coordinatorURI = requireNonNull(coordinatorURI, "coordinatorURI is null");
+        this.httpClient = requireNonNull(httpClient, "httpClient is null");
+        this.taskId = requireNonNull(taskId, "taskId is null");
+        this.source = requireNonNull(source, "source is null");
+        this.driverId = driverId;
+        this.expectedDriversCount = expectedDriversCount;
+    }
+
+    public HttpResponseFuture<JsonResponse<DynamicFilterSummary>> getSummary()
+    {
+        return httpClient.executeAsync(
+                prepareGet()
+                        .setUri(HttpUriBuilder.uriBuilderFrom(coordinatorURI)
+                                .appendPath("/v1/dynamic-filter")
+                                .appendPath("/" + taskId.getQueryId())
+                                .appendPath("/" + source)
+                                .build())
+                        .build(),
+                createFullJsonResponseHandler(jsonCodec(DynamicFilterSummary.class)));
+    }
+
+    public ListenableFuture<StatusResponse> storeSummary(DynamicFilterSummary summary)
+    {
+        return httpClient.executeAsync(
+                preparePut()
+                        .setUri(HttpUriBuilder.uriBuilderFrom(coordinatorURI)
+                                .appendPath("/v1/dynamic-filter")
+                                .appendPath("/" + taskId.getQueryId())
+                                .appendPath("/" + source)
+                                .appendPath("/" + taskId.getStageId().getId())
+                                .appendPath("/" + taskId.getId())
+                                .appendPath("/" + driverId)
+                                .appendPath("/" + expectedDriversCount)
+                                .build())
+                        .addHeader(CONTENT_TYPE, JSON_UTF_8.toString())
+                        .setBodyGenerator(jsonBodyGenerator(summaryJsonCodec, summary))
+                        .build(),
+                createStatusResponseHandler());
+    }
+
+    @Override
+    public void close()
+            throws IOException
+    {
+        // TODO: implement this
```

Comment:
This is a leftover artifact from the early stage of the DPP. We should not have this at all, you're right.